### PR TITLE
SimpleMemoryCheck can now sample memory at a set frequency

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- Added time based memory sampling
- Also added comments in fillDescriptions

#### PR validation:

Code compiles. Made a simple configuration to test the service change and it worked.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Services/plugins/SimpleMemoryCheck.cc [hunk 1]
```diff
@@ -350,6 +355,22 @@ namespace edm {
...
+        if (oncePerEventMode_) {
+          throw cms::Exception("BadConfigOptions")
+              << "'sampleEventNSeconds' and 'oncePerEventMode' cannot be used together";
+        }
+        if (moduleSummaryRequested_) {
+          throw cms::Exception("BadConfigOptions")
+              << "'sampleEventNSeconds' and 'moduleSummaryRequested' cannot be used together";
+        }
+        iReg.watchPostBeginJob(this, &SimpleMemoryCheck::startSamplingThread);
+        iReg.watchPreEndJob(this, &SimpleMemoryCheck::stopSamplingThread);
```
Comment (makortel) [inline FWCore/Services/plugins/SimpleMemoryCheck.cc:368]: Does this need to listen the early termination signals too, or do we call `PreEndJob` after those?

Comment (makortel) [reply FWCore/Services/plugins/SimpleMemoryCheck.cc:368]: I guess with https://github.com/cms-sw/cmssw/pull/45434 we try to call EndJob if it is possible in any way (alternative being the job terminating via `std::terminate()` or similar)?

Comment (Dr15Jones) [reply FWCore/Services/plugins/SimpleMemoryCheck.cc:368]: I agree that the EndJob transition should be run after early termination.

Comment (wddgit) [reply FWCore/Services/plugins/SimpleMemoryCheck.cc:368]: I agree also. That is what is should be doing.

**Comment by makortel** [comment]
Comparison differences are related to https://github.com/cms-sw/cmssw/issues/39803

File: FWCore/Services/plugins/SimpleMemoryCheck.cc [hunk 2]
```diff
@@ -350,6 +355,22 @@ namespace edm {
 
       openFiles();
 
+      if (sampleEveryNSeconds_ > 0) {
+        if (oncePerEventMode_) {
+          throw cms::Exception("BadConfigOptions")
```
Comment (makortel) [inline FWCore/Services/plugins/SimpleMemoryCheck.cc:360]: Should this use
```suggestion
          throw cms::Exception("Configuration")
```
or
```suggestion
          throw edm::Exception(edm::errors::Configuration)
```
?

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
+core

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**